src/main/java/cubicchunks/world/column/Column.java

```diff
@@ -24,136 +24,110 @@
 
+	/**
+	 * Returns Y position of the block directly above the top non-transparent block,
+	 * or null is there are no non-transparent blocks
+	 */
 	@Override
-	@Deprecated
-	// don't use this! It's only here because vanilla needs it
 	public int getHeight(BlockPos pos) {
-		int localX = Coords.blockToLocal(pos.getX());
-		int localZ = Coords.blockToLocal(pos.getZ());
-		return this.getHeightValue(localX, localZ);
+		return this.getHeightValue(
+				Coords.blockToLocal(pos.getX()),
+				Coords.blockToLocal(pos.getZ()));
 	}
 
+	/**
+	 * Returns Y position of the block directly above the top non-transparent block,
+	 * or null is there are no non-transparent blocks
+	 */
 	@Override
-	@Deprecated
-	// don't use this! It's only here because vanilla needs it
 	public int getHeightValue(int localX, int localZ) {
 		// NOTE: the "height value" here is the height of the transparent block
 		// on top of the highest non-transparent block
-
-		Integer skylightBlockY = getHeightmapAt(localX, localZ);
-		if (skylightBlockY == null) {
-			// PANIC! TODO: Test this some more, and find solutions
-			// this column doesn't have any blocks in it that aren't air!
-			// but we can't return null here because vanilla code expects there to be a surface down there somewhere
-			// we don't actually know where the surface is yet, because maybe it hasn't been generated
-			// but we do know that the surface has to be at least at sea level,
-			// so let's go with that for now and hope for the best
-			skylightBlockY = this.getWorld().provider.getAverageGroundLevel() + 1;
-		}
-		return skylightBlockY;
+		return opacityIndex.getTopBlockY(localX, localZ) + 1;
 	}
 
 	@Override
 	@Deprecated //TODO: stop this method form being used by vanila (any algorithms in vanila that use it are be broken any way)
 	// don't use this! It's only here because vanilla needs it
 	public int getTopFilledSegment() {
 		//NOTE: this method actually returns block Y coords
-		Integer cubeY = getTopFilledCubeY();
-		if (cubeY != null) {
-			return Coords.cubeToMinBlock(cubeY);
-		} else {
-			// PANIC!
-			// this column doesn't have any blocks in it that aren't air!
-			// but we can't return null here because vanilla code expects there to be a surface down there somewhere
-			// we don't actually know where the surface is yet, because maybe it hasn't been generated
-			// but we do know that the surface has to be at least at sea level,
-			// so let's go with that for now and hope for the best
-			return this.getWorld().provider.getAverageGroundLevel();
+
+		// PANIC!
+		// this column doesn't have any blocks in it that aren't air!
+		// but we can't return null here because vanilla code expects there to be a surface down there somewhere
+		// we don't actually know where the surface is yet, because maybe it hasn't been generated
+		// but we do know that the surface has to be at least at sea level,
+		// so let's go with that for now and hope for the best
```

Comment:
If it's no longer relevant to the code - yes. If it is still relevant to the code - there is some place for it.